doc/source/changelog.rst

```diff
@@ -1,5 +1,23 @@
 Changelog
 ******************************
+
+- **Unreleased**
```

Comment:
these notes feel a bit developer-oriented. I can't remember, did we decide the changelog should be more user-oriented?

Reply:
Yeah I was tried to emphasize the effect on end use, but it was a little obscured by accompanying implementation detail; what do you think of the updated wording?

Reply:
seems fine with me. Did that particular update actually change anything on the user side? if not, this is fine.

Reply:
Maybe something like "improved error messages" ?

Reply:
Mostly just a couple deprecated property/attribute references, so I'll just let the depr warnings themselves handle that communication I guess. The main user-side thing I wanted to note is with `iterpipes` one, because something like `for p in PipleineInterface` does now behave differently, and _that_ iteration is now via `iterpipes`